lib/Tooling/CrossTranslationUnit.cpp

```diff
@@ -235,6 +235,17 @@ const FunctionDecl *CrossTranslationUnit::getCrossTUDefinition(
     } else {
       Unit = ASTCacheEntry->second.get();
     }
+
+    const auto& LangTo = Context.getLangOpts();
+    const auto& LangFrom = Unit->getASTContext().getLangOpts();
+    // FIXME: Currenty we do not support the import of C AST into C++ or vica
+    // versa. This limitation should be lifted in the future after carefully
+    // examining and handling the cases where the two ASTs are incomatible.
+    if ((LangTo.CPlusPlus && !LangFrom.CPlusPlus) ||
+        ((LangTo.C99 || LangTo.C11) && !(LangFrom.C99 || LangFrom.C11))) {
```

Comment:
Does this handle C89?
`BENIGN_LANGOPT(ImplicitInt, 1, !C99 && !CPlusPlus, "C89 implicit 'int'")`
Also, do we want to enable import from c89 to c11?

Reply:
Do we need this convoluted logic?
I think C11 implies C99 and so on, so each time C11 is true, C99 should be also true.
I was thinking about something like: 
```
if (LangTo.CPlusPlus != LangFrom.CPlusPlus)
```